apps/frontend/src/components/Providers/index.tsx

```diff
@@ -90,42 +108,41 @@ const wagmiConfig = createWagmiConfig({
   },
 });
 
-// For SSR application we need to use
-// createConfig to avoid errors related to window
-// usage inside the connectors.
-const fuelConfig = createConfig(() => ({
-  connectors: defaultConnectors({
-    devMode: appConfig.env === 'testnet',
-    wcProjectId: appConfig.client.walletConnectProjectId,
-    ethWagmiConfig: wagmiConfig,
-    chainId:
-      appConfig.env === 'testnet'
-        ? CHAIN_IDS.fuel.testnet
-        : CHAIN_IDS.fuel.mainnet,
-    fuelProvider: Provider.create(appConfig.client.fuelNodeUrl),
-  }),
-}));
+export const Providers = ({ children }: { children: ReactNode }) => {
+  const fuelProvider = useProvider();
+  const queryClient = getQueryClient();
+
+  useEffect(() => {
+    if (process.env.NODE_ENV === 'development') return;
 
-if (typeof window !== 'undefined' && process.env.NODE_ENV !== 'development') {
-  try {
     posthog.init(process.env.NEXT_PUBLIC_POSTHOG_KEY!, {
       api_host: process.env.NEXT_PUBLIC_POSTHOG_HOST!,
       person_profiles: 'always',
       autocapture: false,
       capture_pageview: true,
       capture_pageleave: false,
-
       loaded: (posthog) => {
         if (process.env.NODE_ENV === 'development') posthog.debug(); // debug mode in development
       },
     });
-  } catch (error) {
-    console.error(error);
-  }
-}
+  }, []);
 
-export const Providers = ({ children }: { children: ReactNode }) => {
-  const queryClient = getQueryClient();
+  const fuelConfig = useMemo(
```

Comment:
1. Do we need this here ?

The configuration was copied from this example: https://github.com/FuelLabs/fuel-connectors/blob/9f18963a0a2330aeb1a8c296a28d7a441d52022b/examples/react-next/src/components/FuelProviders.tsx#L23

2. This setup also causes issues on mainnet:
![image](https://github.com/user-attachments/assets/155262ae-1a3c-4dbf-9255-c148381f1ec7)

3. Is there a problem to do it the old way and initialize 2 providers as we did before ?

Reply:
The error isn't related with the useMemo, everything passwed to our providers should always be either primitive, static, global or memoized